OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -416,7 +421,15 @@ public override bool Tick(Actor self)
 				// Only move by a full speed step if we didn't already move this tick.
 				// If we did, we limit the move to any carried-over leftover progress.
 				if (Move.lastMovePartCompletedTick < self.World.WorldTick)
-					progress += mobile.MovementSpeedForCell(mobile.ToCell);
+				{
+					var currentSpeed = mobile.MovementSpeedForCell(mobile.ToCell);
+					progress += currentSpeed;
+
+					if (mobile.IsTraitDisabled || mobile.IsTraitPaused)
```

Comment:
Would mobile ever be Disabled or Paused here? And if it were, why would we still increment `progress`? I don't think we should have this check here.

Reply:
It is very common in OpenRA-TS to make `Mobile` disabled or paused, for example we use EMP Cannon stop the vehicle moving. If we don't have this check here, after EMP the vehicle still runs at full speed and looks weird (it should first run at initial speed, then full speed). 

The problem is that when  `Mobile` disabled or paused, the actor is still in `Move` activity and thus accelerating (because we lack a mechanic to know the current speed of actor). In other words, in this PR speed is bond to duration of Activity(`Move`) and acceleration, not really use a proper kinematics system to see the current speed.

Reply:
But the check was never there to begin with, so why is it needed now? Also wouldn't `Tick()` on `Move` stop running anyway if `Mobile` is disabled or paused? Since `Move` is called from `Mobile`. I suspect this may be why it doesn't check for it in the existing codebase.

Reply:
> Since Move is called from Mobile.

Activities tick separately

Reply:
Ok so looking at it more we already check for this in `Move`'s `Tick()` and you have even reset the Acceleration as part of this check, so I don't think there is any need to check this again in `MovePart`'s `Tick()`, given that it is already being checked and we have never checked this as part of `MovePart` before.

Reply:
see `Activity.cs`, line 156 in comment.
"Child activities can be queued using QueueChild, and these will be ticked instead of the parent while they are active."

Reply:
`MovePart` is child acitivity of `Move`, so if we want to stop immediately, we have to check both.

Reply:
From what I gather `MovePart` happens once for each cell however, so if you are stopping acceleration during `MovePart` you are stopping in between two cells. This is most likely why there are no checks for `IsTraitPaused` or `IsTraitDisabled` within `MovePart`, because we don't allow units to stop in between cells. So I believe the check should only be in `Move` not `MovePart`.

Reply:
> because we don't allow units to stop in between cells

I don't think so. In TS there is also infantry can be stopped by Reaper's web and stuck in a web. They should not move into cell when stuck in a web, both visually and in game logic. I think it is also for vehicle that being emp.

Reply:
They will still finish their current move step, we know this because there is no exit code in the `Tick()` method. Also please note this variable in the `MovePart` constructor:

![image](https://user-images.githubusercontent.com/1495872/208240234-63a9a615-340b-45c2-9ae8-0ff786435522.png)

Therefore we should not be adding logic to stop acceleration early unless we have a good reason for doing so. If there is already an `IsTraitPaused` or `IsTraitDisabled` check you are free to add your acceleration reset to it, but if no such check exists it should not be added unless there is a special reason for doing so (I see no such reason here).